# TCRBert: Improves Prediction of Epitope-specific TCR recognition using Pre-trained Protein Embeddings.

## Global configurations

In [1]:
import logging
import logging.config
import os
import sys
import warnings
from enum import auto
import pandas as pd

rootdir = '/home/hym/trunk/TCRBert'
workdir = '%s/notebook' % rootdir
datadir = '%s/data' % rootdir
srcdir = '%s/tcrbert' % rootdir
outdir = '%s/output' % rootdir

os.chdir(workdir)

sys.path.append(rootdir)
sys.path.append(srcdir)

# Display
pd.set_option('display.max.rows', 999)
pd.set_option('display.max.columns', 999)

# Logger
warnings.filterwarnings('ignore')
logging.config.fileConfig('../config/logging.conf')
logger = logging.getLogger('tcrbert')

sys.path

['/home/hym/trunk/TCRBert/notebook',
 '/home/hym/trunk/TCRBert/notebook',
 '/home/hym/trunk/epidab',
 '/home/hym/trunk/epidab/epidab',
 '/home/hym/trunk/kvacc',
 '/home/hym/trunk/kvacc/kvacc',
 '/home/hym/trunk/lib',
 '/home/hym/trunk',
 '/home/hym/anaconda3/envs/py37-torch/lib/python37.zip',
 '/home/hym/anaconda3/envs/py37-torch/lib/python3.7',
 '/home/hym/anaconda3/envs/py37-torch/lib/python3.7/lib-dynload',
 '',
 '/home/hym/anaconda3/envs/py37-torch/lib/python3.7/site-packages',
 '/home/hym/anaconda3/envs/py37-torch/lib/python3.7/site-packages/IPython/extensions',
 '/home/hym/.ipython',
 '/home/hym/trunk/TCRBert',
 '/home/hym/trunk/TCRBert/tcrbert']

## Training Datasets

In [2]:
from enum import auto
from tcrbert.commons import StrEnum

class TCREpitopeDFColumnName(StrEnum):
    epitope = auto()
    epitope_gene = auto()
    epitope_species = auto()
    species = auto()
    cdr3b = auto()
    mhc = auto()
    source = auto()
    label = auto()

    @classmethod
    def values(cls):
        return [c.value for c in cls]

def get_index(row, sep='_'):
    return '%s%s%s' % (row[CN.epitope], sep, row[CN.cdr3b])

CN = TCREpitopeDFColumnName


### Postitive dataset

#### Dash data
- containing epitope-specific paired epitope-specific paired TCRα and TCRβ chains for  three epitopes from humans and for seven epitopes from mice[{Dash:2017go}]

In [9]:
import pandas as pd
from IPython.core.display import display
from collections import OrderedDict

gene_info_map = OrderedDict({
    'BMLF': ('EBV', 'GLCTLVAML', 'HLA-A*02:01'),
    'pp65': ('CMV', 'NLVPMVATV', 'HLA-A*02:01'),
    'M1': ('IAV', 'GILGFVFTL', 'HLA-A*02:01'),
    'F2': ('IAV', 'LSLRNPILV', 'H2-Db'),
    'NP': ('IAV', 'ASNENMETM', 'H2-Db'),
    'PA': ('IAV', 'SSLENFRAYV', 'H2-Db'),
    'PB1': ('IAV', 'SSYRRPVGI', 'H2-Kb'),
    'm139': ('mCMV', 'TVYGFCLL', 'H2-Kb'),
    'M38': ('mCMV', 'SSPPMFRV', 'H2-Kb'),
    'M45': ('mCMV', 'HGIRNASFI', 'H2-Db'),   
})

fn_dash = '%s/Dash/human_mouse_pairseqs_v1_parsed_seqs_probs_mq20_clones.tsv' % datadir
df = pd.read_table(fn_dash, sep='\t')
df = df.dropna(subset=['epitope', 'cdr3b'])
df[CN.epitope_gene] = df['epitope']
df[CN.epitope_species] = df[CN.epitope_gene].map(lambda x: gene_info_map[x][0])
df[CN.epitope] = df[CN.epitope_gene].map(lambda x: gene_info_map[x][1])
df[CN.mhc] = df[CN.epitope_gene].map(lambda x: gene_info_map[x][2])
df[CN.species] = df['subject'].map(lambda x: 'human' if 'human' in x else 'mouse')
df[CN.cdr3b] = df['cdr3b'].str.strip().str.upper()
df[CN.source] = 'Dash'
df[CN.label] = 1
df = df.loc[:, CN.values()]
df.index = df.apply(lambda row: get_index(row), axis=1)

df = df[~df.index.duplicated()]
display(df.head(), df.shape)

for i, gene in enumerate(gene_info_map.keys()):
    subdf = df[df[CN.epitope_gene] == gene]
    epitope = subdf[CN.epitope].iat[0]
    species = subdf[CN.epitope_species].iat[0]
    n_cdr3b = subdf.shape[0]
    print('%s: Epitope: %s, species: %s, gene: %s, n_cdr3b: %s' % (i + 1, epitope, species, gene, n_cdr3b))
    
df_dash = df

epitope epitope_gene epitope_species species  \
GLCTLVAML_CASSLWTGSHEQYF  GLCTLVAML         BMLF             EBV   human   
GLCTLVAML_CASSQSPGGEQYF   GLCTLVAML         BMLF             EBV   human   
GLCTLVAML_CASSLTTEQQFF    GLCTLVAML         BMLF             EBV   human   
GLCTLVAML_CSARDATGNGYTF   GLCTLVAML         BMLF             EBV   human   
GLCTLVAML_CSARDSTGNGYTF   GLCTLVAML         BMLF             EBV   human   

                                   cdr3b          mhc source  label  
GLCTLVAML_CASSLWTGSHEQYF  CASSLWTGSHEQYF  HLA-A*02:01   Dash      1  
GLCTLVAML_CASSQSPGGEQYF    CASSQSPGGEQYF  HLA-A*02:01   Dash      1  
GLCTLVAML_CASSLTTEQQFF      CASSLTTEQQFF  HLA-A*02:01   Dash      1  
GLCTLVAML_CSARDATGNGYTF    CSARDATGNGYTF  HLA-A*02:01   Dash      1  
GLCTLVAML_CSARDSTGNGYTF    CSARDSTGNGYTF  HLA-A*02:01   Dash      1

(1408, 8)

1: Epitope: GLCTLVAML, species: EBV, gene: BMLF, n_cdr3b: 54
2: Epitope: NLVPMVATV, species: CMV, gene: pp65, n_cdr3b: 54
3: Epitope: GILGFVFTL, species: IAV, gene: M1, n_cdr3b: 150
4: Epitope: LSLRNPILV, species: IAV, gene: F2, n_cdr3b: 102
5: Epitope: ASNENMETM, species: IAV, gene: NP, n_cdr3b: 149
6: Epitope: SSLENFRAYV, species: IAV, gene: PA, n_cdr3b: 230
7: Epitope: SSYRRPVGI, species: IAV, gene: PB1, n_cdr3b: 335
8: Epitope: TVYGFCLL, species: mCMV, gene: m139, n_cdr3b: 73
9: Epitope: SSPPMFRV, species: mCMV, gene: M38, n_cdr3b: 60
10: Epitope: HGIRNASFI, species: mCMV, gene: M45, n_cdr3b: 201


#### VDJdb

- We extend the positive dataset with new data set from VDJdb{Bagaev:2019hf}](https://vdjdb.cdr3.net, downloaded May 2021), which is a manually curated database that contains TCR sequences with known antigen specificity. Every entry in VDJdb has been given a confidence score between 0  and 3 (0: critical information missing, 1: medium confidence, 2: high confidence, 3: very high  confidence). 
- Confidence score가 죄소한 1이상인 entry를 선택하였다
- MHC-I restricted 에피토프들만 선택하였다.

In [10]:
fn_vdjdb = '%s/VDJdb/vdjdb_20210201.txt' % datadir
################################

df = pd.read_table(fn_vdjdb, sep='\t', header=0)
logger.debug('Current df.shape: %s' % str(df.shape))

# Select beta CDR3 sequence
logger.debug('Select beta CDR3 sequences and MHC-I restricted epitopes')
df = df[(df['gene'] == 'TRB') & (df['mhc.class'] == 'MHCI')]
logger.debug('Current df.shape: %s' % str(df.shape))

# Check valid CDR3 and peptide sequences
logger.debug('Select valid CDR3 and epitope sequences')
df = df.dropna(subset=['cdr3', 'antigen.epitope'])
logger.debug('Current df.shape: %s' % str(df.shape))

logger.debug('Select confidence score > 0')
df = df[df['vdjdb.score'].map(lambda score: score > 0)]
logger.debug('Current df.shape: %s' % str(df.shape))

df[CN.epitope] = df['antigen.epitope'].str.strip().str.upper()
df[CN.epitope_species] = df['antigen.species']
df[CN.epitope_gene] = df['antigen.gene']
df[CN.species] = df['species']
df[CN.cdr3b] = df['cdr3'].str.strip().str.upper()
# df[CN.mhc] = df['mhc.a'].map(lambda x: MHCAlleleName.sub_name(MHCAlleleName.std_name(x)))
df[CN.mhc] = df['mhc.a']
df[CN.source] = 'VDJdb'
df[CN.label] = 1

df.index = df.apply(lambda row: get_index(row), axis=1)
logger.debug('Drop duplicates with the same{epitope, CDR3b}')
df = df[~df.index.duplicated()]
logger.debug('Current df.shape: %s' % str(df.shape))

df = df.loc[:, CN.values()]

# logger.debug('Select all epitope with at least 30 TCRB sequences')
# tmp = df[CN.epitope].value_counts()
# tmp = tmp[tmp >= n_cdr3b_cutoff]
# df = df[df[CN.epitope].map(lambda x: x in tmp.index)]
# logger.debug('Final df.shape: %s' % str(df.shape))

display(df.head(), df.shape)

for i, (epitope, subdf) in enumerate(df.groupby([CN.epitope])):
    epitope_gene = subdf[CN.epitope_gene].unique()
    epitope_species = subdf[CN.epitope_species].iat[0]
    n_cdr3b = subdf.shape[0]

    print(
        '%s: Epitope: %s(%s), speices: %s, gene: %s, n_cdr3b: %s' %
        (i + 1, epitope, len(epitope), epitope_species, epitope_gene, n_cdr3b))

df_vdjdb = df

2021-06-01 03:54:06 [DEBUG]: Current df.shape: (78701, 21)
2021-06-01 03:54:06 [DEBUG]: Select beta CDR3 sequences and MHC-I restricted epitopes
2021-06-01 03:54:06 [DEBUG]: Current df.shape: (44797, 21)
2021-06-01 03:54:06 [DEBUG]: Select valid CDR3 and epitope sequences
2021-06-01 03:54:06 [DEBUG]: Current df.shape: (44797, 21)
2021-06-01 03:54:06 [DEBUG]: Select confidence score > 0
2021-06-01 03:54:06 [DEBUG]: Current df.shape: (6659, 21)
2021-06-01 03:54:06 [DEBUG]: Drop duplicates with the same{epitope, CDR3b}
2021-06-01 03:54:06 [DEBUG]: Current df.shape: (4275, 28)


epitope epitope_gene epitope_species  \
FLKEKGGL_CASSYLPGQGDHYSNQPQHF  FLKEKGGL          Nef           HIV-1   
FLKEKGGL_CASSFEAGQGFFSNQPQHF   FLKEKGGL          Nef           HIV-1   
FLKEKGGL_CASSFEPGQGFYSNQPQHF   FLKEKGGL          Nef           HIV-1   
FLKEKGGL_CASSYEPGQVSHYSNQPQHF  FLKEKGGL          Nef           HIV-1   
FLKEKGGL_CASSALASLNEQFF        FLKEKGGL          Nef           HIV-1   

                                   species                 cdr3b       mhc  \
FLKEKGGL_CASSYLPGQGDHYSNQPQHF  HomoSapiens  CASSYLPGQGDHYSNQPQHF  HLA-B*08   
FLKEKGGL_CASSFEAGQGFFSNQPQHF   HomoSapiens   CASSFEAGQGFFSNQPQHF  HLA-B*08   
FLKEKGGL_CASSFEPGQGFYSNQPQHF   HomoSapiens   CASSFEPGQGFYSNQPQHF  HLA-B*08   
FLKEKGGL_CASSYEPGQVSHYSNQPQHF  HomoSapiens  CASSYEPGQVSHYSNQPQHF  HLA-B*08   
FLKEKGGL_CASSALASLNEQFF        HomoSapiens        CASSALASLNEQFF  HLA-B*08   

                              source  label  
FLKEKGGL_CASSYLPGQGDHYSNQPQHF  VDJdb      1  
FLKEKGGL_CASSFEAGQGFFSNQPQHF   VDJdb      1  
FLKEKGGL_CASSFEPGQGFYSNQPQHF   VDJdb      1  
FLKEKGGL_CASSYEPGQVSHYSNQPQHF  VDJdb      1  
FLKEKGGL_CASSALASLNEQFF        VDJdb      1

(4275, 8)

1: Epitope: AAFKRSCLK(9), speices: MCPyV, gene: ['T-Ag'], n_cdr3b: 5
2: Epitope: AAGIGILTV(9), speices: HomoSapiens, gene: ['MLANA'], n_cdr3b: 2
3: Epitope: ALDPHSGHFV(10), speices: HomoSapiens, gene: ['CDK4'], n_cdr3b: 4
4: Epitope: ALGIGILTV(9), speices: HomoSapiens, gene: ['MLANA'], n_cdr3b: 1
5: Epitope: ALSPVIPHI(9), speices: HomoSapiens, gene: ['MLL2'], n_cdr3b: 2
6: Epitope: ALTPVVVTL(9), speices: HomoSapiens, gene: ['CDK4'], n_cdr3b: 6
7: Epitope: ALWGPDPAAA(10), speices: HomoSapiens, gene: ['INS'], n_cdr3b: 1
8: Epitope: ALYGFVPVL(9), speices: HomoSapiens, gene: ['GANAB'], n_cdr3b: 6
9: Epitope: APARLERRHSA(11), speices: HomoSapiens, gene: ['KIF16B'], n_cdr3b: 3
10: Epitope: APRGPHGGAASGL(13), speices: HomoSapiens, gene: ['NY-ESO-1' 'NY-ESO'], n_cdr3b: 5
11: Epitope: ARMILMTHF(9), speices: HCV, gene: ['NS5B'], n_cdr3b: 14
12: Epitope: ASNENMETM(9), speices: InfluenzaA, gene: ['NP'], n_cdr3b: 69
13: Epitope: ATDALMTGY(9), speices: HCV, gene: ['NS3'], n_cdr3b: 135
14: Epitope: A

#### McPAS-TCR

- McPAS-TCR(http://friedmanlab.weizmann.ac.il/McPAS-TCR/) is a manually curated catalog of pathology-associated TCR sequences[{Tickotsky:2017bo}].
- MHC-I restricted 에피토프들을 선택하였다.

In [5]:
!head -n 1 ../data/McPAS/McPAS-TCR_20210521.csv

"CDR3.alpha.aa","CDR3.beta.aa","Species","Category","Pathology","Pathology.Mesh.ID","Additional.study.details","Antigen.identification.method","Single.cell","NGS","Antigen.protein","Protein.ID","Epitope.peptide","Epitope.ID","MHC","Tissue","T.Cell.Type","T.cell.characteristics","CDR3.alpha.nt","TRAV","TRAJ","TRBV","TRBD","TRBJ","Reconstructed.J.annotation","CDR3.beta.nt","Mouse.strain","PubMed.ID","Remarks"


In [43]:
tokens

['YMLDLQPET', 'KLPQLCTEL']

In [45]:
import numpy as np

fn_mcpas = '%s/McPAS/McPAS-TCR_20210521.csv' % datadir
################################

df = pd.read_csv(fn_mcpas)
logger.debug('Current df.shape: %s' % str(df.shape))

# Select valid beta CDR3 sequence and epitope sequence
logger.debug('Select valid beta CDR3 and epitope sequences')
df = df.dropna(subset=['CDR3.beta.aa', 'Epitope.peptide'])
logger.debug('Current df.shape: %s' % str(df.shape))


# df[CN.epitope] = df['Epitope.peptide'].map(lambda x: x.split('/')[0].upper())
df[CN.epitope] = df['Epitope.peptide'].str.strip().str.upper()

# Handle multiple epitope
logger.debug('Extend by multi-epitopes')
tmpdf = df[df[CN.epitope].str.contains('/')].copy()
for mepitope, subdf in tmpdf.groupby([CN.epitope]):
    logger.debug('Multi epitope: %s' % mepitope)
    tokens = mepitope.split('/')
    logger.debug('Convert epitope: %s to %s' % (mepitope, tokens[0]))
    df[CN.epitope][df[CN.epitope] == mepitope] = tokens[0]    

    for epitope in tokens[1:]:
        logger.debug('Extend by epitope: %s' % epitope)
        subdf[CN.epitope] = epitope
        df = df.append(subdf)
logger.debug('Current df.shape: %s' % (str(df.shape)))
        
df[CN.epitope_gene] = None
df[CN.epitope_species] = df['Pathology']
df[CN.species] = df['Species']
df[CN.cdr3b] = df['CDR3.beta.aa'].str.strip().str.upper()
df[CN.mhc] = df['MHC'].str.strip()
df[CN.source] = 'McPAS'
df[CN.label] = 1

df.index = df.apply(lambda row: get_index(row), axis=1)

logger.debug('Select MHC-I restricted entries')
df = df[
    (df[CN.mhc].notnull()) &
    (np.logical_not(df[CN.mhc].str.contains('DR|DP|DQ')))
]
logger.debug('Current df.shape: %s' % str(df.shape))

logger.debug('Drop duplicates with the same{epitope, CDR3b}')
df = df[~df.index.duplicated()]
logger.debug('Current df.shape: %s' % str(df.shape))

df = df.loc[:, CN.values()]

# logger.debug('Select all epitope with at least 30 TCRB sequences')
# tmp = df[CN.epitope].value_counts()
# tmp = tmp[tmp >= n_cdr3b_cutoff]
# df = df[df[CN.epitope].map(lambda x: x in tmp.index)]
# logger.debug('Final df.shape: %s' % str(df.shape))

display(df.head(), df.shape)
for i, (epitope, subdf) in enumerate(df.groupby([CN.epitope])):
#     epitope_gene = ','.join(subdf[CN.epitope_gene].unique())
    epitope_species = subdf[CN.epitope_species].iat[0]
    n_cdr3b = subdf.shape[0]
    
    print('%s: Epitope: %s(%s), speices: %s, n_cdr3b: %s' % (i + 1, epitope, len(epitope), 
                                                             epitope_species, n_cdr3b))
    
df_mcpas = df

2021-06-01 04:49:26 [DEBUG]: Current df.shape: (39045, 29)
2021-06-01 04:49:26 [DEBUG]: Select valid beta CDR3 and epitope sequences
2021-06-01 04:49:26 [DEBUG]: Current df.shape: (15051, 29)
2021-06-01 04:49:26 [DEBUG]: Extend by multi-epitopes
2021-06-01 04:49:26 [DEBUG]: Multi epitope: YMLDLQPET/KLPQLCTEL
2021-06-01 04:49:26 [DEBUG]: Convert epitope: YMLDLQPET/KLPQLCTEL to YMLDLQPET
2021-06-01 04:49:26 [DEBUG]: Extend by epitope: KLPQLCTEL
2021-06-01 04:49:26 [DEBUG]: Current df.shape: (15096, 30)
2021-06-01 04:49:26 [DEBUG]: Select MHC-I restricted entries
2021-06-01 04:49:26 [DEBUG]: Current df.shape: (13597, 37)
2021-06-01 04:49:26 [DEBUG]: Drop duplicates with the same{epitope, CDR3b}
2021-06-01 04:49:26 [DEBUG]: Current df.shape: (11116, 37)


epitope epitope_gene  \
IKAVYNFATCG_CASSDAGANTEVF   IKAVYNFATCG         None   
IKAVYNFATCG_CASSDAGAYAEQF   IKAVYNFATCG         None   
IKAVYNFATCG_CASSDAGGAAEVF   IKAVYNFATCG         None   
IKAVYNFATCG_CASSDAGHSPLYF   IKAVYNFATCG         None   
IKAVYNFATCG_CASSDAWGGAEQYF  IKAVYNFATCG         None   

                                                      epitope_species species  \
IKAVYNFATCG_CASSDAGANTEVF   Lymphocytic choriomeningitis virus (LCMV)   Mouse   
IKAVYNFATCG_CASSDAGAYAEQF   Lymphocytic choriomeningitis virus (LCMV)   Mouse   
IKAVYNFATCG_CASSDAGGAAEVF   Lymphocytic choriomeningitis virus (LCMV)   Mouse   
IKAVYNFATCG_CASSDAGHSPLYF   Lymphocytic choriomeningitis virus (LCMV)   Mouse   
IKAVYNFATCG_CASSDAWGGAEQYF  Lymphocytic choriomeningitis virus (LCMV)   Mouse   

                                     cdr3b    mhc source  label  
IKAVYNFATCG_CASSDAGANTEVF    CASSDAGANTEVF  H-2db  McPAS      1  
IKAVYNFATCG_CASSDAGAYAEQF    CASSDAGAYAEQF  H-2db  McPAS      1  
IKAVYNFATCG_CASSDAGGAAEVF    CASSDAGGAAEVF  H-2db  McPAS      1  
IKAVYNFATCG_CASSDAGHSPLYF    CASSDAGHSPLYF  H-2db  McPAS      1  
IKAVYNFATCG_CASSDAWGGAEQYF  CASSDAWGGAEQYF  H-2db  McPAS      1

(11116, 8)

1: Epitope: AARAVFLAL(9), speices: Melanoma, n_cdr3b: 2
2: Epitope: ACASQKRPSQR(11), speices: Experimental autoimmune encephalomyelitis (EAE), n_cdr3b: 21
3: Epitope: ALIHHNTHL(9), speices: Neoantigen, n_cdr3b: 4
4: Epitope: ALIHHNTYL(9), speices: Neoantigen, n_cdr3b: 2
5: Epitope: ALLETLSLLL(10), speices: Neoantigen, n_cdr3b: 1
6: Epitope: ALLQVTLLL(9), speices: Neoantigen, n_cdr3b: 1
7: Epitope: ALSPVIPHI(9), speices: Neoantigen, n_cdr3b: 16
8: Epitope: ALSPVIPLI(9), speices: Neoantigen, n_cdr3b: 8
9: Epitope: ALSYTPAEV(9), speices: Neoantigen, n_cdr3b: 3
10: Epitope: ALTPVVVTL(9), speices: Acute myeloid leukemia, n_cdr3b: 6
11: Epitope: ALVGAIPSI(9), speices: Neoantigen, n_cdr3b: 1
12: Epitope: ALWGPDPAA(9), speices: Neoantigen, n_cdr3b: 1
13: Epitope: ALYGFVPVL(9), speices: Neoantigen, n_cdr3b: 24
14: Epitope: ALYGSVPVL(9), speices: Diabetes type 2, n_cdr3b: 7
15: Epitope: AMAGSLVFL(9), speices: Neoantigen, n_cdr3b: 3
16: Epitope: AMAGSPVFL(9), speices: Neoantigen, n_cdr3b: 1
17: E

#### SARS-CoV-2 epitope-specific TCRs from Shomuradova et al

- The SARS-CoV-2 epitope S-protein$_{269-277}$(YLQPRTFLL) with the 352 epitope-specific TCRβs from the recent study of Shomuradova et al[{Shomuradova:2020io}].

In [16]:
fn_shomuradova = '%s/Shomuradova/sars2_tcr.tsv' %datadir

##############################
df = pd.read_csv(fn_shomuradova, sep='\t')
df = df[df['Gene'] == 'TRB']
df[CN.epitope] = df['Epitope'].str.strip().str.upper()
df[CN.epitope_gene] = df['Epitope gene']
df[CN.epitope_species] = df['Epitope species']
df[CN.mhc] = df['MHC A']
df[CN.cdr3b] = df['CDR3'].str.strip().str.upper()
df[CN.species] = df['Species']
df[CN.source] = 'Shomuradova'
df[CN.label] = 1

df.index = df.apply(lambda row: get_index(row), axis=1)
df = df.loc[:, CN.values()]

display(df.head(), df.shape, df[CN.epitope].value_counts())

df_shomuradova = df

epitope epitope_gene epitope_species  \
YLQPRTFLL_CASSFQNTGELFF     YLQPRTFLL        Spike      SARS-CoV-2   
YLQPRTFLL_CASSSVNNNEQFF     YLQPRTFLL        Spike      SARS-CoV-2   
YLQPRTFLL_CAVGEANTGELFF     YLQPRTFLL        Spike      SARS-CoV-2   
YLQPRTFLL_CAYQEVNTGELFF     YLQPRTFLL        Spike      SARS-CoV-2   
YLQPRTFLL_CSARDDQAVNTGELFF  YLQPRTFLL        Spike      SARS-CoV-2   

                                species             cdr3b       mhc  \
YLQPRTFLL_CASSFQNTGELFF     HomoSapiens     CASSFQNTGELFF  HLA-A*02   
YLQPRTFLL_CASSSVNNNEQFF     HomoSapiens     CASSSVNNNEQFF  HLA-A*02   
YLQPRTFLL_CAVGEANTGELFF     HomoSapiens     CAVGEANTGELFF  HLA-A*02   
YLQPRTFLL_CAYQEVNTGELFF     HomoSapiens     CAYQEVNTGELFF  HLA-A*02   
YLQPRTFLL_CSARDDQAVNTGELFF  HomoSapiens  CSARDDQAVNTGELFF  HLA-A*02   

                                 source  label  
YLQPRTFLL_CASSFQNTGELFF     Shomuradova      1  
YLQPRTFLL_CASSSVNNNEQFF     Shomuradova      1  
YLQPRTFLL_CAVGEANTGELFF     Shomuradova      1  
YLQPRTFLL_CAYQEVNTGELFF     Shomuradova      1  
YLQPRTFLL_CSARDDQAVNTGELFF  Shomuradova      1

(381, 8)

YLQPRTFLL    352
RLQSLQTYV     27
VLNDILSRL      1
FIAGLIAIV      1
Name: epitope, dtype: int64

#### Merge positive datasets

-  <strike>After removing duplicates with the same {epitope, CDR3} and selecting all epitope that have at least 30 TCRB CDR3 sequences, the positive dataset contains 12,534 combinations of CDR3b sequences and epitopes with unique 63 epitopes. Table 1 summarizes the positive dataset.</strike>

- After removing duplicates with the same {epitope, CDR3}, the positive dataset contains 14,352 combinations of CDR3b sequences and epitopes with unique 491 epitopes. Table 1 summarizes the positive dataset.

In [19]:
import pandas as pd

n_cdr3b_cutoff = 20
###################################################

logger.debug('Merge all datasets')
df = pd.concat([df_dash, df_vdjdb, df_mcpas, df_shomuradova])
logger.debug('Current df.shape: %s' % str(df.shape))

logger.debug('Drop duplicates')
df = df[~df.index.duplicated()]
logger.debug('Current df.shape: %s' % str(df.shape))

# logger.debug('Select all epitope with at least 30 TCRB sequences')
# tmp = df[CN.epitope].value_counts()
# tmp = tmp[tmp >= n_cdr3b_cutoff]
# df = df[df[CN.epitope].map(lambda x: x in tmp.index)]
# logger.debug('Final df.shape: %s' % str(df.shape))

display(df.head(), df.shape)

for i, (epitope, subdf) in enumerate(df.groupby([CN.epitope])):
    epitope_gene = subdf[CN.epitope_gene].unique()
    source = subdf[CN.source].unique()
    epitope_species = subdf[CN.epitope_species].iat[0]
    n_cdr3b = subdf.shape[0]
    print('%s: Epitope: %s(%s), speices: %s, gene: %s, n_cdr3b: %s, source: %s' % (i + 1, epitope, len(epitope), 
                                                                       epitope_species, epitope_gene, n_cdr3b,
                                                                       source))

df_train_pos = df

2021-06-01 04:02:59 [DEBUG]: Merge all datasets
2021-06-01 04:02:59 [DEBUG]: Current df.shape: (17160, 8)
2021-06-01 04:02:59 [DEBUG]: Drop duplicates
2021-06-01 04:02:59 [DEBUG]: Current df.shape: (14352, 8)


epitope epitope_gene epitope_species species  \
GLCTLVAML_CASSLWTGSHEQYF  GLCTLVAML         BMLF             EBV   human   
GLCTLVAML_CASSQSPGGEQYF   GLCTLVAML         BMLF             EBV   human   
GLCTLVAML_CASSLTTEQQFF    GLCTLVAML         BMLF             EBV   human   
GLCTLVAML_CSARDATGNGYTF   GLCTLVAML         BMLF             EBV   human   
GLCTLVAML_CSARDSTGNGYTF   GLCTLVAML         BMLF             EBV   human   

                                   cdr3b          mhc source  label  
GLCTLVAML_CASSLWTGSHEQYF  CASSLWTGSHEQYF  HLA-A*02:01   Dash      1  
GLCTLVAML_CASSQSPGGEQYF    CASSQSPGGEQYF  HLA-A*02:01   Dash      1  
GLCTLVAML_CASSLTTEQQFF      CASSLTTEQQFF  HLA-A*02:01   Dash      1  
GLCTLVAML_CSARDATGNGYTF    CSARDATGNGYTF  HLA-A*02:01   Dash      1  
GLCTLVAML_CSARDSTGNGYTF    CSARDSTGNGYTF  HLA-A*02:01   Dash      1

(14352, 8)

1: Epitope: AAFKRSCLK(9), speices: MCPyV, gene: ['T-Ag'], n_cdr3b: 5, source: ['VDJdb']
2: Epitope: AAGIGILTV(9), speices: HomoSapiens, gene: ['MLANA'], n_cdr3b: 2, source: ['VDJdb']
3: Epitope: AARAVFLAL(9), speices: Melanoma, gene: [None], n_cdr3b: 2, source: ['McPAS']
4: Epitope: ACASQKRPSQR(11), speices: Experimental autoimmune encephalomyelitis (EAE), gene: [None], n_cdr3b: 21, source: ['McPAS']
5: Epitope: ALDPHSGHFV(10), speices: HomoSapiens, gene: ['CDK4'], n_cdr3b: 4, source: ['VDJdb']
6: Epitope: ALGIGILTV(9), speices: HomoSapiens, gene: ['MLANA'], n_cdr3b: 1, source: ['VDJdb']
7: Epitope: ALIHHNTHL(9), speices: Neoantigen, gene: [None], n_cdr3b: 4, source: ['McPAS']
8: Epitope: ALIHHNTYL(9), speices: Neoantigen, gene: [None], n_cdr3b: 2, source: ['McPAS']
9: Epitope: ALLETLSLLL(10), speices: Neoantigen, gene: [None], n_cdr3b: 1, source: ['McPAS']
10: Epitope: ALLQVTLLL(9), speices: Neoantigen, gene: [None], n_cdr3b: 1, source: ['McPAS']
11: Epitope: ALSPVIPHI(9), speices: Ho

179: Epitope: IVTDFSVIK(9), speices: EBV, gene: ['EBNA4' None], n_cdr3b: 36, source: ['VDJdb' 'McPAS']
180: Epitope: KAFSPEVIPMF(11), speices: HIV-1, gene: ['Gag' None], n_cdr3b: 172, source: ['VDJdb' 'McPAS']
181: Epitope: KASEKIFYV(9), speices: HomoSapiens, gene: ['SSX2'], n_cdr3b: 4, source: ['VDJdb']
182: Epitope: KGYVYQGL(8), speices: VSV, gene: ['N'], n_cdr3b: 9, source: ['VDJdb']
183: Epitope: KIFGSLAFL(9), speices: HomoSapiens, gene: ['ERBB2'], n_cdr3b: 1, source: ['VDJdb']
184: Epitope: KINAWIKVV(9), speices: HIV-1, gene: ['Gag'], n_cdr3b: 3, source: ['VDJdb']
185: Epitope: KIVALGINAV(10), speices: Neoantigen, gene: [None], n_cdr3b: 4, source: ['McPAS']
186: Epitope: KLAKPLPYT(9), speices: Neoantigen, gene: [None], n_cdr3b: 1, source: ['McPAS']
187: Epitope: KLANPLPYT(9), speices: Neoantigen, gene: [None], n_cdr3b: 2, source: ['McPAS']
188: Epitope: KLEFMDWRL(9), speices: synthetic, gene: ['synthetic'], n_cdr3b: 1, source: ['VDJdb']
189: Epitope: KLEQLDWTV(9), speices: HomoSap

489: Epitope: YTSEILSPI(9), speices: Neoantigen, gene: [None], n_cdr3b: 1, source: ['McPAS']
490: Epitope: YVLDHLIVV(9), speices: EBV, gene: ['BRLF1' None], n_cdr3b: 92, source: ['VDJdb' 'McPAS']
491: Epitope: YVLTWIVGA(9), speices: Neoantigen, gene: [None], n_cdr3b: 6, source: ['McPAS']


In [20]:
14352 * 2

28704

### Negative dataset

- To increase the specificity of the predictive model, negative examples wich we do not expect to interact between TCRs and epitopes were added.
- The negative examples were made by combining the peptides from the positive dataset with the randomly selected background TCR CDR3 beta sequences constructed by Dash et al[{Dash:2017go}] which have been collected from two healty donors[{Howie:2015dc}]
- 예측 모델의 학습에서의 Overfitting을 피하기 위해, trainging 데이터셋은 epitope 별로 같은 수의 positive/negative data point를 포함하도록 하였다. 최종 Training 데이터셋은 28,704개의 TCR CDR3b-peptide combinations를 포함하고 있다(Table S1)

In [21]:
fn_cntr = '%s/TCRGP/human_tcr_control.csv' % datadir
################################
df_cntr = pd.read_csv(fn_cntr)
display(df_cntr.head())
pos_cdr3b = df_train_pos[CN.cdr3b].unique()
neg_cdr3b = list(filter(lambda x: x not in pos_cdr3b, df_cntr['cdr3b'].unique()))

print('len(pos_cdr3b): %s, len(neg_cdr3b): %s' % (len(pos_cdr3b), len(neg_cdr3b)))

df = pd.DataFrame(columns=CN.values())
for epitope, subdf in df_train_pos.groupby([CN.epitope]):
    subdf_neg = subdf.copy()
    subdf_neg[CN.source] = 'Control'
    subdf_neg[CN.label] = 0
    subdf_neg[CN.cdr3b] = np.random.choice(neg_cdr3b, subdf.shape[0], replace=False)
    subdf_neg.index = subdf_neg.apply(lambda row: get_index(row), axis=1)
    df = df.append(subdf_neg)

display(df.head(), df.shape)    
df_train_neg = df

va_reps                      vb_reps             cdr3a  \
0     TRAV9-2*01                   TRBV7-9*01   CALSPGGTSYGKLTF   
1     TRAV9-2*01  TRBV20/OR9-2*01;TRBV20-1*01    CALKGYTGRRALTF   
2     TRAV9-2*01                   TRBV5-1*01     CALSTGANSKLTF   
3       TRAV6*01                    TRBV27*01  CALDMRSGAGSYQLTF   
4  TRAV29/DV5*01                   TRBV7-6*01    CAAPFKGGSEKLVF   

              cdr3b  
0  CASSPATGGLVDTQYF  
1   CSAREGLAGDYEQYF  
2   CASRRALAGADEQYF  
3   CASRPFQVSTDTQYF  
4    CASSYSPHNSPLHF

len(pos_cdr3b): 13715, len(neg_cdr3b): 8039


epitope epitope_gene epitope_species  \
AAFKRSCLK_CASSSGRFERAAHDTQYF  AAFKRSCLK         T-Ag           MCPyV   
AAFKRSCLK_CASSLAGPSSAQHF      AAFKRSCLK         T-Ag           MCPyV   
AAFKRSCLK_CASSFGLAGGPNTGELFF  AAFKRSCLK         T-Ag           MCPyV   
AAFKRSCLK_CASSSTGVSYEQYF      AAFKRSCLK         T-Ag           MCPyV   
AAFKRSCLK_CASRTGLGEQYF        AAFKRSCLK         T-Ag           MCPyV   

                                  species               cdr3b          mhc  \
AAFKRSCLK_CASSSGRFERAAHDTQYF  HomoSapiens  CASSSGRFERAAHDTQYF  HLA-A*03:01   
AAFKRSCLK_CASSLAGPSSAQHF      HomoSapiens      CASSLAGPSSAQHF  HLA-A*03:01   
AAFKRSCLK_CASSFGLAGGPNTGELFF  HomoSapiens  CASSFGLAGGPNTGELFF  HLA-A*03:01   
AAFKRSCLK_CASSSTGVSYEQYF      HomoSapiens      CASSSTGVSYEQYF  HLA-A*03:01   
AAFKRSCLK_CASRTGLGEQYF        HomoSapiens        CASRTGLGEQYF  HLA-A*03:01   

                               source label  
AAFKRSCLK_CASSSGRFERAAHDTQYF  Control     0  
AAFKRSCLK_CASSLAGPSSAQHF      Control     0  
AAFKRSCLK_CASSFGLAGGPNTGELFF  Control     0  
AAFKRSCLK_CASSSTGVSYEQYF      Control     0  
AAFKRSCLK_CASRTGLGEQYF        Control     0

(14352, 8)

In [22]:
fn_train = '%s/train.csv' % outdir

df_train = pd.concat([df_train_pos, df_train_neg])

display(df_train.head(), df_train.shape)
print('Duplicated count: %s' % np.count_nonzero(df_train.index.duplicated()))
max_len = df_train.index.map(lambda x: len(x) - 1).max()
print('max_len: %s' % max_len)
df_train.to_csv(fn_train)

epitope epitope_gene epitope_species species  \
GLCTLVAML_CASSLWTGSHEQYF  GLCTLVAML         BMLF             EBV   human   
GLCTLVAML_CASSQSPGGEQYF   GLCTLVAML         BMLF             EBV   human   
GLCTLVAML_CASSLTTEQQFF    GLCTLVAML         BMLF             EBV   human   
GLCTLVAML_CSARDATGNGYTF   GLCTLVAML         BMLF             EBV   human   
GLCTLVAML_CSARDSTGNGYTF   GLCTLVAML         BMLF             EBV   human   

                                   cdr3b          mhc source label  
GLCTLVAML_CASSLWTGSHEQYF  CASSLWTGSHEQYF  HLA-A*02:01   Dash     1  
GLCTLVAML_CASSQSPGGEQYF    CASSQSPGGEQYF  HLA-A*02:01   Dash     1  
GLCTLVAML_CASSLTTEQQFF      CASSLTTEQQFF  HLA-A*02:01   Dash     1  
GLCTLVAML_CSARDATGNGYTF    CSARDATGNGYTF  HLA-A*02:01   Dash     1  
GLCTLVAML_CSARDSTGNGYTF    CSARDSTGNGYTF  HLA-A*02:01   Dash     1

(28704, 8)

Duplicated count: 0
max_len: 42


### Independent external dataset

#### SARS-Cov-2 epitopes from ImmuneCODE

- The SARS-CoV-2 epitope S-protein$_{269-277}$(YLQPRTFLL) with the 451 epitope-specific TCRβs from the ImmuneRACE study launched by Adaptive Biotechnologies and Microsoft (https://immunerace.adaptivebiotech.com, June 10, 2020 dataset)

In [23]:
fn_ic = '%s/ImmuneCODE/sars2_YLQPRTFLL_with_neg.csv' % datadir


df = pd.read_csv(fn_ic)

df[CN.epitope] = 'YLQPRTFLL'
df[CN.epitope_gene] = 'Spike'
df[CN.epitope_species] = 'SARS-CoV-2'
df[CN.mhc] = None
df[CN.cdr3b] = df['cdr3b'].str.strip().str.upper()
df[CN.species] = 'human'
df[CN.source] = 'ImmuneCODE'
df[CN.label] = df['subject'].map(lambda x: 0 if x == 'control' else 1)

df.index = df.apply(lambda row: get_index(row), axis=1)
df = df.loc[:, CN.values()]

display(df.head(), df.shape, df.label.value_counts())

df_immunecode = df

epitope epitope_gene epitope_species species  \
YLQPRTFLL_CAKGLANTGELFF     YLQPRTFLL        Spike      SARS-CoV-2   human   
YLQPRTFLL_RKQWQGEETQYF      YLQPRTFLL        Spike      SARS-CoV-2   human   
YLQPRTFLL_CASSYRNTGELFF     YLQPRTFLL        Spike      SARS-CoV-2   human   
YLQPRTFLL_CSARDNWAVNTGELFF  YLQPRTFLL        Spike      SARS-CoV-2   human   
YLQPRTFLL_CASSPDIEAFF       YLQPRTFLL        Spike      SARS-CoV-2   human   

                                       cdr3b   mhc      source  label  
YLQPRTFLL_CAKGLANTGELFF        CAKGLANTGELFF  None  ImmuneCODE      1  
YLQPRTFLL_RKQWQGEETQYF          RKQWQGEETQYF  None  ImmuneCODE      1  
YLQPRTFLL_CASSYRNTGELFF        CASSYRNTGELFF  None  ImmuneCODE      1  
YLQPRTFLL_CSARDNWAVNTGELFF  CSARDNWAVNTGELFF  None  ImmuneCODE      1  
YLQPRTFLL_CASSPDIEAFF            CASSPDIEAFF  None  ImmuneCODE      1

(830, 8)

0    415
1    415
Name: label, dtype: int64

## PyTorch Dataset Class

In [ ]:
!pip install tape_proteins

import tape

tape.__version__

In [24]:
import torch
from torch.utils.data import Dataset
import numpy as np
from tape import TAPETokenizer

class TCREpitopeSentenceDataset(Dataset):
    
    def __init__(self, df=None, max_len=None):
        self._df = df
        self.max_len = max_len

    def __getitem__(self, index):
        
        row = self._df.iloc[index, :]
        sequence_ids = eval(row['sequence'])
        label = row[CN.label]
        
        return torch.tensor(sequence_ids), torch.tensor(label)

    def __len__(self):
        return self._df.shape[0]
    
    @classmethod
    def from_data_file(cls, fn=None, max_len=None, tokenizer=None):
        def encode_row(row):
            epitope = row[CN.epitope]
            cdr3b = row[CN.cdr3b]
            logger.debug('Encoding epitope: %s, cdr3b: %s' % (epitope, cdr3b))
            sequence_ids = tokenizer.encode(epitope)
            
            sequence_ids = np.append(sequence_ids, tokenizer.encode(cdr3b))
            n_pads = max_len - sequence_ids.shape[0]
            if n_pads > 0:
                sequence_ids = np.append(sequence_ids, [tokenizer.vocab['<pad>']]*n_pads)
            return sequence_ids
            
        df = pd.read_csv(fn)
        df['sequence'] = df.apply(encode_row, axis=1)
        
        return cls(df=df, max_len=max_len)
        
        
##########################################
tokenizer = TAPETokenizer(vocab='iupac')
train_ds = TCREpitopeSentenceDataset.from_data_file(fn=fn_train, max_len=max_len, tokenizer=tokenizer)

2021-06-01 04:06:01 [INFO]: Better speed can be achieved with apex installed from https://www.github.com/nvidia/apex .
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: GLCTLVAML, cdr3b: CASSLWTGSHEQYF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: GLCTLVAML, cdr3b: CASSQSPGGEQYF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: GLCTLVAML, cdr3b: CASSLTTEQQFF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: GLCTLVAML, cdr3b: CSARDATGNGYTF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: GLCTLVAML, cdr3b: CSARDSTGNGYTF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: GLCTLVAML, cdr3b: CSARDRTGNGYTF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: GLCTLVAML, cdr3b: CSARGQGDEQYF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: GLCTLVAML, cdr3b: CASSSGAGLDEQFF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: GLCTLVAML, cdr3b: CASSQSPGGTQYF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: GLCTLVAML, cdr3b: CASGGSSYQETQYF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: GLCTLVAML, cdr3b: CSVGSAGTNEKLFF
2021-06-01

2021-06-01 04:06:01 [DEBUG]: Encoding epitope: GILGFVFTL, cdr3b: CASSGGRNQPQHF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: GILGFVFTL, cdr3b: CASSPLIGEQYF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: GILGFVFTL, cdr3b: CSARVRGDVYGYTF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: GILGFVFTL, cdr3b: CASSQVEAFYEQYF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: GILGFVFTL, cdr3b: CASSLERNTYEQYF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: GILGFVFTL, cdr3b: CASSQDPGYEQYF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: GILGFVFTL, cdr3b: CASSYSGHVNTEAFF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: GILGFVFTL, cdr3b: CSARDGTRAHYEQFF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: GILGFVFTL, cdr3b: CASSIFGGAEAFF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: GILGFVFTL, cdr3b: CASSQDWGAGTQYF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: NLVPMVATV, cdr3b: CATEQGMHEQYF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: NLVPMVATV, cdr3b: CASSPITGTGAYGYTF
2021-06-01 04:06:01 [DEBUG]: Encoding epito

2021-06-01 04:06:01 [DEBUG]: Encoding epitope: GILGFVFTL, cdr3b: CASSIRSGVEQYF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: GILGFVFTL, cdr3b: CASSILTGPRTEAFF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: GILGFVFTL, cdr3b: CASSFRSTDTQYF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: GILGFVFTL, cdr3b: CASSSRSANEQFF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: GILGFVFTL, cdr3b: CASSPRASLEQYF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: GILGFVFTL, cdr3b: CASSIAPTDTQYF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: NLVPMVATV, cdr3b: CAVNGGSNYGYTF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: NLVPMVATV, cdr3b: CAWSVSDVANTEAFF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: NLVPMVATV, cdr3b: CASSYFDEKLFF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: GLCTLVAML, cdr3b: CASSQSPGGQQFF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: GLCTLVAML, cdr3b: CSVGSDGTNEKLFF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: GLCTLVAML, cdr3b: CSARVEPGNGYTF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: G

2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSPPMFRV, cdr3b: CASGGNYAEQFF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSPPMFRV, cdr3b: CASSPGTGGYNYAEQFF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSPPMFRV, cdr3b: CATGTGSSAETLYF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSPPMFRV, cdr3b: CASSGGTGGFNYAEQFF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSPPMFRV, cdr3b: CTCSDGTGGYNYAEQFF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSPPMFRV, cdr3b: CTCSPGTGGVNYAEQFF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSYRRPVGI, cdr3b: CSSSQSSAETLYF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSYRRPVGI, cdr3b: CASSLPDSQNTLYF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSYRRPVGI, cdr3b: CASSGQSNERLFF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSYRRPVGI, cdr3b: CASSFPDSAETLYF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSYRRPVGI, cdr3b: CASSQDSAETLYF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSYRRPVGI, cdr3b: CASKGRTSAETLYF
2021-06-01 04:06:01 [DEBUG]: Encoding e

2021-06-01 04:06:01 [DEBUG]: Encoding epitope: ASNENMETM, cdr3b: CASSLTAISNERLFF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: ASNENMETM, cdr3b: CASSPWGGADQNTLYF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: ASNENMETM, cdr3b: CASGEDWGKDTQYF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: ASNENMETM, cdr3b: CASSRDWGDAEQFF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: ASNENMETM, cdr3b: CASSQDLGGRYEQYF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: ASNENMETM, cdr3b: CASSQGLGGRGEQYF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: ASNENMETM, cdr3b: CLSIQDRRVSYNSPLYF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSYRRPVGI, cdr3b: CASSGDSSAETLYF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSYRRPVGI, cdr3b: CASSYSSAETLYF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSYRRPVGI, cdr3b: CASSSDSAETLYF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSYRRPVGI, cdr3b: CASSLDSAETLYF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: ASNENMETM, cdr3b: CASSQEDWGSAETLYF
2021-06-01 04:06:01 [DEBUG]: Enco

2021-06-01 04:06:01 [DEBUG]: Encoding epitope: LSLRNPILV, cdr3b: CASSDDRVTNQAPLF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSPPMFRV, cdr3b: CASSLATGGYNYAEQFF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSPPMFRV, cdr3b: CASSRDNWSDYTF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSPPMFRV, cdr3b: CASRDNNQDTQYF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSLENFRAYV, cdr3b: CASSGGGEQYF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSLENFRAYV, cdr3b: CASAEGGEQYF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSLENFRAYV, cdr3b: CASSWGDTLYF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSLENFRAYV, cdr3b: CASTPDRGEVFF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSLENFRAYV, cdr3b: CASSQAEVFF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSLENFRAYV, cdr3b: CASSLGDEQYF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSLENFRAYV, cdr3b: CASSIGGEQYF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSLENFRAYV, cdr3b: CASSPTGGDQNTLYF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSLENF

2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSLENFRAYV, cdr3b: CASASGGEVFF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSLENFRAYV, cdr3b: CASSWGGEVFF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSLENFRAYV, cdr3b: CASSPDITEVFF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSLENFRAYV, cdr3b: CASSGGETLYF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSLENFRAYV, cdr3b: CASSSGGGQFF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSLENFRAYV, cdr3b: CAWSPGREVFF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSLENFRAYV, cdr3b: CASSLGNEQYF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSLENFRAYV, cdr3b: CASSWGTEVFF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSYRRPVGI, cdr3b: CASRGDQSAETLYF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSYRRPVGI, cdr3b: CASSPDWGAETLYF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSYRRPVGI, cdr3b: CAWTSQSAETLYF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSYRRPVGI, cdr3b: CASTTGSDERLFF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSYRRPVGI,

2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSLENFRAYV, cdr3b: CASSELGGQGEQYF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSLENFRAYV, cdr3b: CASSQDRNTLYF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSLENFRAYV, cdr3b: CASSSGDF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSLENFRAYV, cdr3b: CASSPYEQYF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSLENFRAYV, cdr3b: CASSWGEEVFF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSLENFRAYV, cdr3b: CASREGGFSNERLFF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSLENFRAYV, cdr3b: CASSQGDEQYF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSLENFRAYV, cdr3b: CASSPDRGEQYF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSLENFRAYV, cdr3b: CASTQGAEVFF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSLENFRAYV, cdr3b: CASSSGQEVFF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSLENFRAYV, cdr3b: CASSLGGPTEVFF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSYRRPVGI, cdr3b: CASTAGDSAETLYF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSYRRPVG

2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSLENFRAYV, cdr3b: CASSFGAEQFF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSLENFRAYV, cdr3b: CASSFGQAPLF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSLENFRAYV, cdr3b: CASSPTGGGNERLFF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSLENFRAYV, cdr3b: CASSLPEGPTGQLYF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSLENFRAYV, cdr3b: CASSLNRGRLYF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSLENFRAYV, cdr3b: CASSPDRGRLFF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSLENFRAYV, cdr3b: CASSLTGEQYF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSLENFRAYV, cdr3b: CASSSGQAPLF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSLENFRAYV, cdr3b: CASSSGEAPLF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSLENFRAYV, cdr3b: CASSWGVEQYF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSLENFRAYV, cdr3b: CASTGGQAPLF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSLENFRAYV, cdr3b: CASSEGERLFF
2021-06-01 04:06:01 [DEBUG]: Encoding epitope: SSLENFR

2021-06-01 04:06:02 [DEBUG]: Encoding epitope: SSLENFRAYV, cdr3b: CASSGGGEVFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: SSLENFRAYV, cdr3b: CASTWGGEQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: SSLENFRAYV, cdr3b: CASSREQDRGWEQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: SSLENFRAYV, cdr3b: CASSLSGTDFSNERLFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: SSLENFRAYV, cdr3b: CASSEGQPNSDYTF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: SSLENFRAYV, cdr3b: CASSQEADSYEQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: SSLENFRAYV, cdr3b: CASSLGSEQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: SSLENFRAYV, cdr3b: CASSYGAEQFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: SSLENFRAYV, cdr3b: CASSIGERLFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: SSLENFRAYV, cdr3b: CASSSPGEQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: SSYRRPVGI, cdr3b: CASSSWGGLSQNTLYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: SSYRRPVGI, cdr3b: CASSIVGGGAETLYF
2021-06-01 04:06:02 [DEBUG]: Encoding epi

2021-06-01 04:06:02 [DEBUG]: Encoding epitope: SSYRRPVGI, cdr3b: CASSDDWETLYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: SSYRRPVGI, cdr3b: CASSERDWGGEQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: SSYRRPVGI, cdr3b: CTCSADRENTEVFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: SSYRRPVGI, cdr3b: CTCSADRINQDTQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: SSYRRPVGI, cdr3b: CTCSGRTSAETLYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: SSYRRPVGI, cdr3b: CASSDDTYAEQFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: SSYRRPVGI, cdr3b: CASSQEDTSNERLFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: SSYRRPVGI, cdr3b: CASSLGTDQDTQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: SSYRRPVGI, cdr3b: CAIRATSAETLYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: SSYRRPVGI, cdr3b: CASSLFPGQANSDYTF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: SSYRRPVGI, cdr3b: CASSDAPTSNERLFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: SSYRRPVGI, cdr3b: CASSNLEAEQFF
2021-06-01 04:06:02 [DEBUG]: Encoding epi

2021-06-01 04:06:02 [DEBUG]: Encoding epitope: SSYRRPVGI, cdr3b: CASSDGTEVFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: SSYRRPVGI, cdr3b: CASSWDSQNTLYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: SSYRRPVGI, cdr3b: CASGDPGIQDTQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: SSYRRPVGI, cdr3b: CATTGSSAETLYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: SSYRRPVGI, cdr3b: CACGGTDSAETLYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: SSYRRPVGI, cdr3b: CAWTGFSNERLFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: SSYRRPVGI, cdr3b: CASSTGGAYEQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: SSYRRPVGI, cdr3b: CASSLHRANTEVFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: SSYRRPVGI, cdr3b: CASNGGSSAETLYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: SSYRRPVGI, cdr3b: CASGDREEQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: SSYRRPVGI, cdr3b: CASRGHSSAETLYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: SSYRRPVGI, cdr3b: CASSSPDSAETLYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: SSY

2021-06-01 04:06:02 [DEBUG]: Encoding epitope: HGIRNASFI, cdr3b: CASSYRGDTEVFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: HGIRNASFI, cdr3b: CASSLMGAPYEQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: HGIRNASFI, cdr3b: CASSQLGTSAETLYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: HGIRNASFI, cdr3b: CASSDQQGLSAETLYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: HGIRNASFI, cdr3b: CASSLRERAETLYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: HGIRNASFI, cdr3b: CASSLRQGNTEVFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: HGIRNASFI, cdr3b: CASSFSGLYEQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: HGIRNASFI, cdr3b: CASSLAREDTQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: HGIRNASFI, cdr3b: CASSLEDWGKDTQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: HGIRNASFI, cdr3b: CASSFRGDQNTLYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: HGIRNASFI, cdr3b: CASSSPGLGDYAEQFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: HGIRNASFI, cdr3b: CASSHLGSYEQYF
2021-06-01 04:06:02 [DEBUG]: Encoding 

2021-06-01 04:06:02 [DEBUG]: Encoding epitope: HGIRNASFI, cdr3b: CASSRDWGGQDTQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: LSLRNPILV, cdr3b: CASSDGGAGNTLYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: LSLRNPILV, cdr3b: CASSSGRGDTQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: LSLRNPILV, cdr3b: CASSPGQYYAEQFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: LSLRNPILV, cdr3b: CASSPGASSDYTF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: LSLRNPILV, cdr3b: CASSPGTGGWAPLF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: LSLRNPILV, cdr3b: CASSWGLGNYAEQFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: LSLRNPILV, cdr3b: CASSIGDEQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: LSLRNPILV, cdr3b: CASDLGGKDTQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: LSLRNPILV, cdr3b: CASSFGREQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: LSLRNPILV, cdr3b: CASSDARLGVDTQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: LSLRNPILV, cdr3b: CAWSAYGDNYAEQFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope

2021-06-01 04:06:02 [DEBUG]: Encoding epitope: KRWIILGLNK, cdr3b: CASSLVAGTYNEQFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: KRWIILGLNK, cdr3b: CSASDNELPGSSYNEQFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: KRWIILGLNK, cdr3b: CASSMADLSYEQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: KRWIILGLNK, cdr3b: CASSGGLPEGYTF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: KRWIILGLNK, cdr3b: CSASNPLLGQGAGYGYTF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: KRWIILGLNK, cdr3b: CSGRGVAYNSPLHF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: KRWIILGLNK, cdr3b: CASSQDITFPKQFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: KRWIILGLNK, cdr3b: CSVPGLAQNEQFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: KRWIILGLNK, cdr3b: CASSPGTGELFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: KRWIILGLNK, cdr3b: CASSQGQYSHEQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: KRWIILGLNK, cdr3b: CASSQGRLSAEAFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: KRWIILGLNK, cdr3b: CASSPGRLGNEQYF
2021-06-01 04:06:02 [DE

2021-06-01 04:06:02 [DEBUG]: Encoding epitope: KRWIIMGLNK, cdr3b: CASSLLGGGGEAFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: KRWIIMGLNK, cdr3b: CASSFSSLGEQFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: KRWIILGLNK, cdr3b: CSAPLAGAPQDTQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: KRWIILGLNK, cdr3b: CASRPGQGSHEQFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: KRWIILGLNK, cdr3b: CASSFGPSNQPQHF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: KRWIIMGLNK, cdr3b: CASSLLEHNNGELFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: KRWIIMGLNK, cdr3b: CASRPGQGSHEQFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: NLVPMVATV, cdr3b: CASSKQSGTGYYGYTF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: NLVPMVATV, cdr3b: CASSQLGPSAYEQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: NLVPMVATV, cdr3b: CSARDQRGLAGGIGDTQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: NLVPMVATV, cdr3b: CASSLQTGAGGFGYTF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: NLVPMVATV, cdr3b: CASSAVAGAPLEQYF
2021-06-01 04:06:02 [

2021-06-01 04:06:02 [DEBUG]: Encoding epitope: LLWNGPMAV, cdr3b: CASSDSGTDTQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: LLWNGPMAV, cdr3b: CASSFLAGGNEQFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: LLWNGPMAV, cdr3b: CAWSGDRAMVSGELFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: LLWNGPMAV, cdr3b: CASSEGANTEAFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: LLWNGPMAV, cdr3b: CSASRRGGTNEKLFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: LLWNGPMAV, cdr3b: CASRKTGYTGELFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: LLWNGPMAV, cdr3b: CASSMRRGTDTQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: LLWNGPMAV, cdr3b: CASSHWREESEQFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: LLWNGPMAV, cdr3b: CSARDAPSRETQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: LLLGIGILV, cdr3b: CASGGTGATYSNQPQHF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: LLLGIGILV, cdr3b: CASRPDFTGEQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: LLLGIGILV, cdr3b: CASSPTSIANTGELFF
2021-06-01 04:06:02 [DEBUG]: Encodi

2021-06-01 04:06:02 [DEBUG]: Encoding epitope: EIYKRWII, cdr3b: CASSVVGDFRETQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: EIYKRWII, cdr3b: CASSYGGTEAFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: EIYKRWII, cdr3b: CASSLPGQGRTPLHF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: EIYKRWII, cdr3b: CASSFLPKNEQFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: EIYKRWII, cdr3b: CSARLSYEQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: EIYKRWII, cdr3b: CASSAITSGGARDEQFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: EIYKRWII, cdr3b: CASSVGGDHREEQHF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: EIYKRWII, cdr3b: CATIRTGFSSYEQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: QVPLRPMTYK, cdr3b: CASSYSRGSGNTIYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: QVPLRPMTYK, cdr3b: CAISAGASFVTRSTDTQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: QVPLRPMTYK, cdr3b: CASRQQGFVFEAKNIQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: QVPLRPMTYK, cdr3b: CSAPTSGSAAFF
2021-06-01 04:06:02 [DEBUG]: Encodi

2021-06-01 04:06:02 [DEBUG]: Encoding epitope: GTSGSPIIDK, cdr3b: CAIRAGTATYGYTF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: GTSGSPIIDK, cdr3b: CASSGNEQFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: GTSGSPIIDK, cdr3b: CASSLKGGWAYNEQFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: GTSGSPIIDK, cdr3b: CASSQRSRDRGAFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: GTSGSPIIDK, cdr3b: CASSYTFRDTKETQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: GTSGSPIINR, cdr3b: CASSWPEETQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: GTSGSPIINR, cdr3b: CASSLGGGNQETQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: GTSGSPIINR, cdr3b: CASSFGLAGVYRY
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: GTSGSPIINR, cdr3b: CASSGGAAYEQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: GTSGSPIINR, cdr3b: CASSPSQDTEAFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: GTSGSPIIDK, cdr3b: CASSGGQGNIQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: GTSGSPIIDK, cdr3b: CASSLIGGGNTEAFF
2021-06-01 04:06:02 [DEBUG]: Enc

2021-06-01 04:06:02 [DEBUG]: Encoding epitope: KLVALGINAV, cdr3b: CASSYGSYEQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: KLVALGINAV, cdr3b: CASSWGGGEQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: KLVALGINAV, cdr3b: CASSLEGSYEQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: KLVALGINAV, cdr3b: CSAIDFPMGNEQFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: KLVALGINAV, cdr3b: CASSSRHEGEDTEAFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: KLVALGINAV, cdr3b: CASREDGASGSPDTQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: KLVALGINAV, cdr3b: CASKMGAEAFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: KLVALGINAV, cdr3b: CASSLGGDIQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: KLVALGINAV, cdr3b: CASTGGLGYTF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: KLVALGINAV, cdr3b: CASSYQSGNTEAFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: KLVALGINAV, cdr3b: CASNRGDGYTF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: KLVALGINAV, cdr3b: CASSIALSDNYGYTF
2021-06-01 04:06:02 [DEBUG]: Encoding 

2021-06-01 04:06:02 [DEBUG]: Encoding epitope: GILGFVFTL, cdr3b: CASSQGAYGYTF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: GILGFVFTL, cdr3b: CASSLLLARETQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: GILGFVFTL, cdr3b: CASSRRSSDEQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: GILGFVFTL, cdr3b: CASSLLGGWSEAFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: GILGFVFTL, cdr3b: CASSIRSSTEAFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: NLVPMVATV, cdr3b: CASSEDHTGFYEQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: NLVPMVATV, cdr3b: CASSPQTGVGNYGYTF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: GILGFVFTL, cdr3b: CASSLIYPGELFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: KRWIILGLNK, cdr3b: CASSLLVSGNEQFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: KRWIILGLNK, cdr3b: CASSTRQNTEAFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: KRWIILGLNK, cdr3b: CASSQGVGNTIYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: KRWIILGLNK, cdr3b: CASSRYQTEAFF
2021-06-01 04:06:02 [DEBUG]: Encoding epit

2021-06-01 04:06:02 [DEBUG]: Encoding epitope: YVLDHLIVV, cdr3b: CASSQGFGGGETQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: RAKFKQLL, cdr3b: CASSQGVGENTEAFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: RAKFKQLL, cdr3b: CASSQVRGRGEQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: AVFDRKSDAK, cdr3b: CASSRTSATGELFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: FLRGRAYGL, cdr3b: CASSSGQAYEQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: YVLDHLIVV, cdr3b: CASSSLRGSNQPQHF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: YVLDHLIVV, cdr3b: CASSSLTGSGQPQHF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: RAKFKQLL, cdr3b: CASSSTEQATYGYTF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: FLRGRAYGL, cdr3b: CASSVGQAYEQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: RAKFKQLL, cdr3b: CASSVLAGDTQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: IVTDFSVIK, cdr3b: CASSWGGGSNYGYTF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: IVTDFSVIK, cdr3b: CASSWGQGSNYGYTF
2021-06-01 04:06:02 [DEBUG]: Encoding e

2021-06-01 04:06:02 [DEBUG]: Encoding epitope: RAKFKQLL, cdr3b: CASSPITRIVGDTQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: RAKFKQLL, cdr3b: CASSPLIGTSGTEAFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: YVLDHLIVV, cdr3b: CASSPLPGTTGTDTQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: RAKFKQLL, cdr3b: CASSPPGPTGRGWYDTF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: RAKFKQLL, cdr3b: CASSPSKPGDNEQFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: YVLDHLIVV, cdr3b: CASSPSLVGSADTQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: VSFIEFVGW, cdr3b: CASSPTGTSHTGELFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: RPPIFIRRL, cdr3b: CASSPTNPAGDTETQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: RAKFKQLL, cdr3b: CASSQAGQGNSPLHF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: RAKFKQLL, cdr3b: CASSQAQGMGNSPLHF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: AVFDRKSDAK, cdr3b: CASSQEGTGSYEQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: RPPIFIRRL, cdr3b: CASSQETGSASQHF
2021-06-01 04:06:02 [DE

2021-06-01 04:06:02 [DEBUG]: Encoding epitope: VTEHDTLLY, cdr3b: CASSVGQGVTYEQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: CTELKLSDY, cdr3b: CASSPGLAGVYEQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: VTEHDTLLY, cdr3b: CASRLAGGGGEQFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: VTEHDTLLY, cdr3b: CATSSLSGRVTGELFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: VTEHDTLLY, cdr3b: CSVRTGGLGANVLTF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: VTEHDTLLY, cdr3b: CASSQERGGKWAYEQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: VTEHDTLLY, cdr3b: CASSVGQGKTYEQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: VTEHDTLLY, cdr3b: CATHRAGGFNEQFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: NLVPMVATV, cdr3b: CASSSETTSPLHF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: GILGFVFTL, cdr3b: CASSMFVGQPQHF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: GILGFVFTL, cdr3b: CASSYYSNQPQHF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: NLVPMVATV, cdr3b: CASSVVERAFF
2021-06-01 04:06:02 [DEBUG]: Encoding

2021-06-01 04:06:02 [DEBUG]: Encoding epitope: GLCTLVAML, cdr3b: CSARGGTGNGYTF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: GLCTLVAML, cdr3b: CSARDRTGNTIYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: GLCTLVAML, cdr3b: CSARSEVGNTIYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: GLCTLVAML, cdr3b: CSAHRTENTEAFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: GLCTLVAML, cdr3b: CSVGGTFYGYTF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: GLCTLVAML, cdr3b: CSVPGTDYNEQFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: GLCTLVAML, cdr3b: CSVDLANTEAFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: GLCTLVAML, cdr3b: CSVGQGGTNEKLFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: GLCTLVAML, cdr3b: CSVGSGGTNERLFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: GLCTLVAML, cdr3b: CSVEASGGLFNEQFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: GLCTLVAML, cdr3b: CSATDSMNTEAFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: GLCTLVAML, cdr3b: CSVPGQVLETQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: GLC

2021-06-01 04:06:02 [DEBUG]: Encoding epitope: GPGHKARVL, cdr3b: CATSRLDITPNSPLHF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: GPGHKARVL, cdr3b: CATSREEPIQPGEQFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: GPGHKARVL, cdr3b: CATRSTDTQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: GPGHKARVL, cdr3b: CATSRVGQWNTEAFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: GPGHKARVL, cdr3b: CATSSRGGETQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: GPGHKARVL, cdr3b: CATGRLAGETQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: GPGHKARVL, cdr3b: CATSREPGSIVNEQFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: SLYNTVATL, cdr3b: CASSFDVEQFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: SLYNTVATL, cdr3b: CASSPRTEKYTQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: SLYNTVATL, cdr3b: CASLSGGAFYHTGELFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: SLYNTVATL, cdr3b: CASSYSTHYSLESDTQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: SLYNTVATL, cdr3b: CASSEARWVAGVWTGELFF
2021-06-01 04:06:02 [DEBUG]: 

2021-06-01 04:06:02 [DEBUG]: Encoding epitope: RALEYKNL, cdr3b: CASSLGGGNSDYTF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: RALEYKNL, cdr3b: CASSLDSGPSYEQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: RALEYKNL, cdr3b: CASSLDSGDTEVFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: RALEYKNL, cdr3b: CASRRQLSNERLFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: RALEYKNL, cdr3b: CASSLGWLNSDYTF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: RALEYKNL, cdr3b: CTCSGTGDTEVFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: RALEYKNL, cdr3b: CASRRTGGRDEQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: RALEYKNL, cdr3b: CASSLNRMNAEQFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: RALEYKNL, cdr3b: CAWSLGGRGEQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: RALEYKNL, cdr3b: CASSRDWGDYEQYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: RALEYKNL, cdr3b: CASSPGQGLFF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: RALEYKNL, cdr3b: CASGDVQGDTLYF
2021-06-01 04:06:02 [DEBUG]: Encoding epitope: RALEYKNL, 

2021-06-01 04:06:03 [DEBUG]: Encoding epitope: TPRVTGGGAM, cdr3b: CASSLGAAGRELFF
2021-06-01 04:06:03 [DEBUG]: Encoding epitope: TPRVTGGGAM, cdr3b: CASSPRDSQPYGYTF
2021-06-01 04:06:03 [DEBUG]: Encoding epitope: TPRVTGGGAM, cdr3b: CASSKEDREKETQYF
2021-06-01 04:06:03 [DEBUG]: Encoding epitope: TPRVTGGGAM, cdr3b: CASSLGSANTDTQYF
2021-06-01 04:06:03 [DEBUG]: Encoding epitope: TPRVTGGGAM, cdr3b: CSGGALGQDFGGYTF
2021-06-01 04:06:03 [DEBUG]: Encoding epitope: TPRVTGGGAM, cdr3b: CASSLSYSTLSYEQYF
2021-06-01 04:06:03 [DEBUG]: Encoding epitope: TPRVTGGGAM, cdr3b: CATAEGAGELFF
2021-06-01 04:06:03 [DEBUG]: Encoding epitope: TPRVTGGGAM, cdr3b: CASSLSHSTGNYGYTF
2021-06-01 04:06:03 [DEBUG]: Encoding epitope: TPRVTGGGAM, cdr3b: CASRLGLAGGINEQFF
2021-06-01 04:06:03 [DEBUG]: Encoding epitope: TPRVTGGGAM, cdr3b: CASTLGSAGPQFNEQFF
2021-06-01 04:06:03 [DEBUG]: Encoding epitope: TPRVTGGGAM, cdr3b: CASSLSPSTGRVEQYF
2021-06-01 04:06:03 [DEBUG]: Encoding epitope: TPRVTGGGAM, cdr3b: CASSLGSTPYEQYF
2021-06-01 04:0

2021-06-01 04:06:03 [DEBUG]: Encoding epitope: ALTPVVVTL, cdr3b: CASSSGGGNEQFF
2021-06-01 04:06:03 [DEBUG]: Encoding epitope: ALTPVVVTL, cdr3b: CASSAQGFGELFF
2021-06-01 04:06:03 [DEBUG]: Encoding epitope: ALTPVVVTL, cdr3b: CASSEGRETQYF
2021-06-01 04:06:03 [DEBUG]: Encoding epitope: ALTPVVVTL, cdr3b: CASSIGQGPYEQYF
2021-06-01 04:06:03 [DEBUG]: Encoding epitope: ALTPVVVTL, cdr3b: CASSLGGRRYEQYF
2021-06-01 04:06:03 [DEBUG]: Encoding epitope: ALTPVVVTL, cdr3b: CASSLGQKEQYF
2021-06-01 04:06:03 [DEBUG]: Encoding epitope: GILGFVFTL, cdr3b: CASSIRSVAEAFF
2021-06-01 04:06:03 [DEBUG]: Encoding epitope: GILGFVFTL, cdr3b: CASSVSATGELFF
2021-06-01 04:06:03 [DEBUG]: Encoding epitope: GILGFVFTL, cdr3b: CASSHRGTGELFF
2021-06-01 04:06:03 [DEBUG]: Encoding epitope: GILGFVFTL, cdr3b: CASSLRSTGELFF
2021-06-01 04:06:03 [DEBUG]: Encoding epitope: GILGFVFTL, cdr3b: CASSDRSSQPQHF
2021-06-01 04:06:03 [DEBUG]: Encoding epitope: CINGVCWTV, cdr3b: CASSDPLAGGNEQYF
2021-06-01 04:06:03 [DEBUG]: Encoding epitope: CIN

2021-06-01 04:06:03 [DEBUG]: Encoding epitope: NEGVKAAW, cdr3b: CASSLGQLEQYF
2021-06-01 04:06:03 [DEBUG]: Encoding epitope: NEGVKAAW, cdr3b: CASSLGQIEQYF
2021-06-01 04:06:03 [DEBUG]: Encoding epitope: NEGVKAAW, cdr3b: CASSLGQLEAFF
2021-06-01 04:06:03 [DEBUG]: Encoding epitope: NEGVKAAW, cdr3b: CASTERGVLTDTQYF
2021-06-01 04:06:03 [DEBUG]: Encoding epitope: NEGVKAAW, cdr3b: CASSVQLPLGVPYEQYF
2021-06-01 04:06:03 [DEBUG]: Encoding epitope: NEGVKAAW, cdr3b: CANSRGGSDNEQFF
2021-06-01 04:06:03 [DEBUG]: Encoding epitope: NEGVKAAW, cdr3b: CASSIVAGSYNEQFF
2021-06-01 04:06:03 [DEBUG]: Encoding epitope: NEGVKAAW, cdr3b: CASSIVTGPYNEQFF
2021-06-01 04:06:03 [DEBUG]: Encoding epitope: NEGVKAAW, cdr3b: CASSISGLNTDTQYF
2021-06-01 04:06:03 [DEBUG]: Encoding epitope: NEGVKAAW, cdr3b: CASSFWTGHADTQYF
2021-06-01 04:06:03 [DEBUG]: Encoding epitope: NEGVKAAW, cdr3b: CASSLRGVWTDTQYF
2021-06-01 04:06:03 [DEBUG]: Encoding epitope: NEGVKAAW, cdr3b: CASSIVGEQFF
2021-06-01 04:06:03 [DEBUG]: Encoding epitope: NEGVK

IOPub message rate exceeded.
The notebook server will temporarily stop sending output
to the client in order to avoid crashing it.
To change this limit, set the config variable
`--NotebookApp.iopub_msg_rate_limit`.

Current values:
NotebookApp.iopub_msg_rate_limit=1000.0 (msgs/sec)
NotebookApp.rate_limit_window=3.0 (secs)



2021-06-01 04:06:03 [DEBUG]: Encoding epitope: RPHERNGFTVL, cdr3b: CASSPNRNTEAFF
2021-06-01 04:06:03 [DEBUG]: Encoding epitope: RPHERNGFTVL, cdr3b: CASSPGRNTEAFF
2021-06-01 04:06:03 [DEBUG]: Encoding epitope: RPHERNGFTVL, cdr3b: CASSPTRNTEAFF
2021-06-01 04:06:03 [DEBUG]: Encoding epitope: TPRVTGGGAM, cdr3b: CASSFRTVSSYEQYF
2021-06-01 04:06:03 [DEBUG]: Encoding epitope: TPRVTGGGAM, cdr3b: CASSFRTVNSYEQYF
2021-06-01 04:06:03 [DEBUG]: Encoding epitope: TPRVTGGGAM, cdr3b: CASSLRTVSSYEQYF
2021-06-01 04:06:03 [DEBUG]: Encoding epitope: TPRVTGGGAM, cdr3b: CASSLKGVSSYNEQFF
2021-06-01 04:06:03 [DEBUG]: Encoding epitope: TPRVTGGGAM, cdr3b: CASSFRQGVNTGELFF
2021-06-01 04:06:03 [DEBUG]: Encoding epitope: TPRVTGGGAM, cdr3b: CASSFRQGSNTGELFF
2021-06-01 04:06:03 [DEBUG]: Encoding epitope: TPRVTGGGAM, cdr3b: CASSFRSGINTGELFF
2021-06-01 04:06:03 [DEBUG]: Encoding epitope: TPRVTGGGAM, cdr3b: CASSFRQGTPTGELFF
2021-06-01 04:06:03 [DEBUG]: Encoding epitope: TPRVTGGGAM, cdr3b: CASSSRGTVNTEAFF
2021-06-01 04:

2021-06-01 04:06:04 [DEBUG]: Encoding epitope: VVVGADGVGK, cdr3b: CASSLTDPLDSDYTF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: KAFSPEVIPMF, cdr3b: CASSSLVNTGELF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: KAFSPEVIPMF, cdr3b: CASRKGQGDWEAFF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: KAFSPEVIPMF, cdr3b: CASSLIAGGGEDTQYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: KAFSPEVIPMF, cdr3b: CASGGANYGYTF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: KAFSPEVIPMF, cdr3b: CASSFWGQGTDTQYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: KAFSPEVIPMF, cdr3b: CASSQGVTTGGFLNYGYTF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: KAFSPEVIPMF, cdr3b: CASTDTGYYGYTF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: CTPYDINQM, cdr3b: CASSLGIRQFTEAFF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: CTPYDINQM, cdr3b: CASSFLLQVLGKLFF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: CTPYDINQM, cdr3b: CASSEQGNSGNTVYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: CTPYDINQM, cdr3b: CASSETGNSNQPQYF
2021-06-01 04:06

2021-06-01 04:06:04 [DEBUG]: Encoding epitope: STPESANL, cdr3b: CASSLTRVSNQPQYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: STPESANL, cdr3b: CASSLNRLGNQPQYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: STPESANL, cdr3b: CASSPNRLSNQPQYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: STPESANL, cdr3b: CASSWDRRGNQPQYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: STPESANL, cdr3b: CASSKSRLSNQPQYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: STPESANL, cdr3b: CASSGRDRAYTEAFF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: STPESANL, cdr3b: CASSTSRLSNQPQYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: STPESANL, cdr3b: CASSIARGSQNTQYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: STPESANL, cdr3b: CASSLFRGSNQPQYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: STPESANL, cdr3b: CASSLGRNSNQPQYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: STPESANL, cdr3b: CASSQDKNRAGNQPQYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: STPESANL, cdr3b: CAITVMDLNYDYTF
2021-06-01 04:06:04 [DEBUG]: Encoding e

2021-06-01 04:06:04 [DEBUG]: Encoding epitope: CTPYDINQM, cdr3b: CASSERGNSQNTQYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: CTPYDINQM, cdr3b: CASSGGGNSNQPQYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: CTPYDINQM, cdr3b: CASSEGTGSNQPQYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: CTPYDINQM, cdr3b: CASSENSNSNQPQYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: CTPYDINQM, cdr3b: CASSEAVRRSGASVLTF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: CTPYDINQM, cdr3b: CASSEALAVRNTVYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: CTPYDINQM, cdr3b: CASSLAPGQGGSNQPQYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: CTPYDINQM, cdr3b: CASSLSIRASGASVLTF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: CTPYDINQM, cdr3b: CASSEARRQTDPQYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: CTPYDINQM, cdr3b: CASSDTGNSNQPQYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: CTPYDINQM, cdr3b: CASSYSGTERNTVYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: CTPYDINQM, cdr3b: CASRQNRGSSAQLFF
2021-06-01 04:06:04 [

2021-06-01 04:06:04 [DEBUG]: Encoding epitope: TTPESANL, cdr3b: CASSLSRMSNQPQYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: TTPESANL, cdr3b: CASSQGRQSNQPQYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: TTPESANL, cdr3b: CASSIDRGSQNTQYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: TTPESANL, cdr3b: CASSQERFGTSNEKLFF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: TTPESANL, cdr3b: CASSPQGDNSPLYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: TTPESANL, cdr3b: CSAKSEFRNQPQYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: TTPESANL, cdr3b: CASSQEKVSNQPQYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: TTPESANL, cdr3b: CASSQEVGRLGGRTAQLFF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: TTPESANL, cdr3b: CASAPNRGSNQPQYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: TTPESANL, cdr3b: CASSQERAGGTANSPLYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: TTPESANL, cdr3b: CASSQERQGLSNQPQYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: TTPESANL, cdr3b: CASSPDRVSNQPQYF
2021-06-01 04:06:04 [DEBUG]: En

2021-06-01 04:06:04 [DEBUG]: Encoding epitope: NLSALGIFST, cdr3b: CASGLDIHAFF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: NLSALGIFST, cdr3b: CASSPLDNWEQFF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: NLSALGIFST, cdr3b: CASSLEVTYEQYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: NLSALGIFST, cdr3b: CASSLAGEGGNTIYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: NLSALGIFST, cdr3b: CASSEGQNYGYTF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: NLSALGIFST, cdr3b: CATSDRGQGANWDEQFF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: NLSALGIFST, cdr3b: CASSVATLRVSTNEKLFF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: NLSALGIFST, cdr3b: CASSIQRNYGYTF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: NLSALGIFST, cdr3b: CASSATFNDNEKLFF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: NLSALGIFST, cdr3b: CATSDFGTEAFF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: NLSALGIFST, cdr3b: CASSQGQLAGGLVF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: NLSALGIFST, cdr3b: CASSGSKGAAGELFF
2021-06-01 04:06:04 [DEBUG]

2021-06-01 04:06:04 [DEBUG]: Encoding epitope: SQLLNAKYL, cdr3b: CASSDWGVRDTQYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: SQLLNAKYL, cdr3b: CASSDWGAGAEQFF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: IKAVYNFATCG, cdr3b: CASSDAGANTEVF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: IKAVYNFATCG, cdr3b: CASSDAGAYAEQF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: IKAVYNFATCG, cdr3b: CASSDAGGAAEVF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: IKAVYNFATCG, cdr3b: CASSDAGHSPLYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: IKAVYNFATCG, cdr3b: CASSDAWGGAEQYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: IKAVYNFATCG, cdr3b: CASSDGANTEVF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: IKAVYNFATCG, cdr3b: CASSDNANSDYTF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: IKAVYNFATCG, cdr3b: CASSDQANTEVF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: IKAVYNFATCG, cdr3b: CASSDSANTEVF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: IKAVYNFATCG, cdr3b: CASSDSLNTEVF
2021-06-01 04:06:04 [DEBUG]: Enco

2021-06-01 04:06:04 [DEBUG]: Encoding epitope: VITAFTEGLK, cdr3b: CASASGTGNTLYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: VITAFTEGLK, cdr3b: CASGRQNQRLF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: VITAFNGELK, cdr3b: CASGRQNQRLF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: VITAFTEGLK, cdr3b: CASSCRRPGISNERLF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: VITAFTEGLK, cdr3b: CASSDGQGNQNTLYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: VITAFTEGLK, cdr3b: CASSGGRDNSYEQYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: VITAFTEGLK, cdr3b: CASSPRTGINTGQLYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: VITAFNGELK, cdr3b: CASSPRTGINTGQLYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: VITAFTEGLK, cdr3b: CASSQEKGLGGREQYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: VITAFTEGLK, cdr3b: CASSQGGGDTQYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: VITAFTEGLK, cdr3b: CASSQGQGAGQNTLYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: VITAFNGELK, cdr3b: CASSQGQGAGQNTLYF
2021-06-01 04:06:04 [D

2021-06-01 04:06:04 [DEBUG]: Encoding epitope: ASNENMETM, cdr3b: CARGAATTEVF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: ASNENMETM, cdr3b: CARGGANTGQLYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: ASNENMETM, cdr3b: CARGGGNTGQLYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: ASNENMETM, cdr3b: CARGGGRTGQLYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: ASNENMETM, cdr3b: CARGGNTGQLYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: ASNENMETM, cdr3b: CARGGSNTGQLYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: ASNENMETM, cdr3b: CASARDGNYAEQF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: ASNENMETM, cdr3b: CASARTANTEVF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: ASNENMETM, cdr3b: CASARTGNTEVF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: ASNENMETM, cdr3b: CASDLAGNAEQF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: ASNENMETM, cdr3b: CASDWGRDEQF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: ASNENMETM, cdr3b: CASDWQGRGNTLVF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: ASNENMETM, c

2021-06-01 04:06:04 [DEBUG]: Encoding epitope: ASNENMETM, cdr3b: CASNWGGDTGQLYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: ASNENMETM, cdr3b: CASSAGTAEVF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: ASNENMETM, cdr3b: CASSARANTGQLTEVF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: ASNENMETM, cdr3b: CASSARTANTEVF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: ASNENMETM, cdr3b: CASSARVDNQAPQHF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: ASNENMETM, cdr3b: CASSCAGSNTGQHF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: ASNENMETM, cdr3b: CASSDAGGQDTQYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: ASNENMETM, cdr3b: CASSDAGGTEVF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: ASNENMETM, cdr3b: CASSDLTGKAGNTLYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: ASNENMETM, cdr3b: CASSDPGGYNSPLHF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: ASNENMETM, cdr3b: CASSDPTGGYAEQF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: ASNENMETM, cdr3b: CASSDRQANTEVF
2021-06-01 04:06:04 [DEBUG]: Encoding epi

2021-06-01 04:06:04 [DEBUG]: Encoding epitope: LSLRNPILV, cdr3b: CASSIVTEVF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: LSLRNPILV, cdr3b: CASSLGANERLF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: LSLRNPILV, cdr3b: CASSLGDYAEQF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: LSLRNPILV, cdr3b: CASSLGLGKEQYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: LSLRNPILV, cdr3b: CASSMEGSSAETLYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: LSLRNPILV, cdr3b: CASSMGGIYAEQF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: LSLRNPILV, cdr3b: CASSMGSNERLF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: LSLRNPILV, cdr3b: CASSMGTGNSDYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: LSLRNPILV, cdr3b: CASSMGTGVETLYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: LSLRNPILV, cdr3b: CASSMGVLDTQYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: LSLRNPILV, cdr3b: CASSPDISYNSPLHF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: LSLRNPILV, cdr3b: CASSPGAIPEVF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: LSLRNPI

2021-06-01 04:06:04 [DEBUG]: Encoding epitope: EAAGIGILTV, cdr3b: CASSQQGLGTEAF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: EAAGIGILTV, cdr3b: CASSQYEVQETQYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: EAAGIGILTV, cdr3b: CASSVQGLGQPQHF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: EAAGIGILTV, cdr3b: CASSVWADPNEAF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: EAAGIGILTV, cdr3b: CASSWSQLVHEQF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: EAAGIGILTV, cdr3b: CASSYGDLYTGELF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: EAAGIGILTV, cdr3b: CASSYRGQSNQPQHF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: EAAGIGILTV, cdr3b: CASTLTGLGQPQHF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: EAAGIGILTV, cdr3b: CASTPPGSETQYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: EAAGIGILTV, cdr3b: CASTYTSLAHNEQF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: EAAGIGILTV, cdr3b: CATAPREDRVGNEAF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: EAAGIGILTV, cdr3b: CATSDQGGTEDEQF
2021-06-01 04:06:04 [DEBUG]: E

2021-06-01 04:06:04 [DEBUG]: Encoding epitope: EAAGIGILTV, cdr3b: CASSFGGMGQPQHF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: EAAGIGILTV, cdr3b: CASSFGKIAGDEQYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: EAAGIGILTV, cdr3b: CASSFLGSSSYEQYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: EAAGIGILTV, cdr3b: CASSFLGTASYEQYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: EAAGIGILTV, cdr3b: CASSFMGSSSYEQYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: EAAGIGILTV, cdr3b: CASSFMGTSSYEQYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: EAAGIGILTV, cdr3b: CASSFNTGELF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: EAAGIGILTV, cdr3b: CASSFQGLGQPQHF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: EAAGIGILTV, cdr3b: CASSFQGLGSPLHF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: EAAGIGILTV, cdr3b: CASSFTVGGQETQYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: EAAGIGILTV, cdr3b: CASSGAGGWTEAF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: EAAGIGILTV, cdr3b: CASSGTGLGQPQHF
2021-06-01 04:06:04 [DEBUG

2021-06-01 04:06:04 [DEBUG]: Encoding epitope: EAAGIGILTV, cdr3b: CASSVDAGGAGELF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: EAAGIGILTV, cdr3b: CASSVDVGGLEAF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: EAAGIGILTV, cdr3b: CASSVEGGGAWTDTQYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: EAAGIGILTV, cdr3b: CASSVGFAGDEAF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: EAAGIGILTV, cdr3b: CASSVGGLAIGELF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: EAAGIGILTV, cdr3b: CASSVGGLNTGELF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: EAAGIGILTV, cdr3b: CASSVGQAAEAF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: EAAGIGILTV, cdr3b: CASSVVPTVGFAGELF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: EAAGIGILTV, cdr3b: CASSWDGTLGQPQHF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: EAAGIGILTV, cdr3b: CASSWGGTSGNTIYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: EAAGIGILTV, cdr3b: CASSWGQGQETQYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: EAAGIGILTV, cdr3b: CASSWVGQGAFEAF
2021-06-01 04:06:04 [DEBU

2021-06-01 04:06:04 [DEBUG]: Encoding epitope: MEVGWYRPPFSRVVHLYRNGK, cdr3b: CASGETGGVYAEQF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: ELAGIGILTV, cdr3b: CASRLATLVGNEQF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: ELAGIGILTV, cdr3b: CASRSGSLNTEAF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: ELAGIGILTV, cdr3b: CASRWGQLTNYGYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: ELAGIGILTV, cdr3b: CASSENQLNMTEAF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: ELAGIGILTV, cdr3b: CASSIAAVSALYGYTLF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: ELAGIGILTV, cdr3b: CASSKPSGGINEQF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: ELAGIGILTV, cdr3b: CASSLQGLGTEAF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: ELAGIGILTV, cdr3b: CASSLTTPPSTDPQYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: ELAGIGILTV, cdr3b: CASSPGQGLPNHF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: ELAGIGILTV, cdr3b: CASSQGQLSSYNEQF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: ELAGIGILTV, cdr3b: CASSQQGLGTEAF
2021-06-01 04:0

2021-06-01 04:06:04 [DEBUG]: Encoding epitope: NLVPMVATV, cdr3b: CASTSWAQASTDTQYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: NLVPMVATV, cdr3b: CASTVDSLDTEAF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: NLVPMVATV, cdr3b: CATSDLGQGLDNEQF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: NLVPMVATV, cdr3b: CSAPDLFGQGIETQYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: NLVPMVATV, cdr3b: CSAQSGLAGGSLQETQYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: NLVPMVATV, cdr3b: CSARDAGLAGLNGAPSSYEQYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: NLVPMVATV, cdr3b: CSARDPFGTGYLNSNQPQHF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: NLVPMVATV, cdr3b: CSAREPDDTREAF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: NLVPMVATV, cdr3b: CSARGDTGTGYEQYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: NLVPMVATV, cdr3b: CSARHDLGQGYEQYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: NLVPMVATV, cdr3b: CSARTTYGTDIISQHF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: NLVPMVATV, cdr3b: CSARVVPGVQETQYF
2021-06-01 04:

2021-06-01 04:06:04 [DEBUG]: Encoding epitope: GLCTLVAML, cdr3b: CSVPGTDYNEQF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: GLCTLVAML, cdr3b: CSVPSLLNTEAF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: GLCTLVAML, cdr3b: CSVQEGGYGYTF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: RAKFKQLL, cdr3b: CSVTAGEDDTQYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: RAKFKQLL, cdr3b: CSVTAGEDQPQHF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: RAKFKQLL, cdr3b: CSVTSERTGTTWGYTF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: RAKFKQLL, cdr3b: CSYSPGQGRNEQF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: TQGYFPDWQNY, cdr3b: CAISECAIRGTSGLTDTQYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: TQGYFPDWQNY, cdr3b: CAISERAIRGTSGLTDTQYL
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: EIYKRWII, cdr3b: CASELSGNTIYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: FLKEKGGL, cdr3b: CASKEELSNTGELF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: FLKEKGGL, cdr3b: CASRGGSGELF
2021-06-01 04:06:04 [DEBUG]: Encoding epito

2021-06-01 04:06:04 [DEBUG]: Encoding epitope: KRWIILGLNK, cdr3b: CASSQGQLGNTIYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: KRWIILGLNK, cdr3b: CASSQGQYGNEQF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: KRWIILGLNK, cdr3b: CASSQPPDRGYGYTF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: KRWIILGLNK, cdr3b: CASSRTQGPNTDTQYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: KRWIILGLNK, cdr3b: CASSSKTGELF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: KRWIILGLNK, cdr3b: CASSSRTGELF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: KRWIILGLNK, cdr3b: CASSSTRGTEAF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: KRWIILGLNK, cdr3b: CASSTTSGRYEQYF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: KRWIILGLNK, cdr3b: CASSVLGTSGGAEQF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: KRWIILGLNK, cdr3b: CATSETGELF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: KRWIILGLNK, cdr3b: CATSGVTGELF
2021-06-01 04:06:04 [DEBUG]: Encoding epitope: KRWIILGLNK, cdr3b: CAWSLTGMNQPQHF
2021-06-01 04:06:04 [DEBUG]: Encoding ep

KeyError: "Unrecognized token: '/'"